### PR TITLE
uncompress zip files when source URL points to one

### PR DESCRIPTION
**Pull request by attwad**
Advantage is that it requires no configuration per source, disadvantage is that it makes assumptions like having one file in the zip file that we're interested in. Fortunately I think that's the case for Mexico data that we're trying to ingest so it seems fine to me. I couldn't confirm though as I can't download the original zip file (need to vpn somewhere it seems, can't DL from Europe :( ).

For #1155

### CODE REVIEW

**Comment by axmb** [comment]
> Advantage is that it requires no configuration per source, disadvantage is that it makes assumptions like having one file in the zip file that we're interested in. Fortunately I think that's the case for Mexico data that we're trying to ingest so it seems fine to me. I couldn't confirm though as I can't download the original zip file (need to vpn somewhere it seems, can't DL from Europe :( ).
> 
> For #1155

It's indeed a single CSV file.

File: ingestion/functions/retrieval/retrieval.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import codecs
+import io
 import json
+import mimetypes
 import os
 import sys
-import io
+import tempfile
+import zipfile
```
Comment (axmb) [inline ingestion/functions/retrieval/retrieval.py:8]: Sanity checking -- these are all python3 and don't require declaration in `requirements.txt`, right?

File: ingestion/functions/retrieval/retrieval.py [hunk 2]
```diff
@@ -74,6 +77,25 @@ def get_source_details(env, source_id, upload_id, api_headers, cookies):
             e, env, common_lib.UploadError.INTERNAL_ERROR, source_id, upload_id,
             api_headers, cookies)
 
+def raw_content(url:str, content: bytes) -> io.BytesIO:
```
Comment (axmb) [inline ingestion/functions/retrieval/retrieval.py:80]: hyper-nit: `url: str`

File: ingestion/functions/retrieval/retrieval_test.py [hunk 3]
```diff
@@ -373,3 +376,24 @@ def test_upload_to_s3_raises_error_on_s3_error(
 
     # We got the wrong exception or no exception, fail the test.
     assert not "Should have raised an exception."
+
+def test_decompress_zip_file():
```
Comment (axmb) [inline ingestion/functions/retrieval/retrieval_test.py:380]: nit, but `test_raw_content_unzips` and `test_raw_content_ignores_unknown_mimetypes`?

**Review by axmb — APPROVED** [review]

File: ingestion/functions/retrieval/retrieval.py (hunk 1 above)
Comment (attwad) [reply ingestion/functions/retrieval/retrieval.py:8]: yup, python has "battery included" as they like to say

File: ingestion/functions/retrieval/retrieval.py (hunk 2 above)
Comment (attwad) [reply ingestion/functions/retrieval/retrieval.py:80]: omg how can you... okay done ;)

File: ingestion/functions/retrieval/retrieval_test.py (hunk 3 above)
Comment (attwad) [reply ingestion/functions/retrieval/retrieval_test.py:380]: done

**Pull request merged**